programs/ziti-edge-tunnel/instance-config.c

```diff
@@ -0,0 +1,158 @@
+/*
+ Copyright 2019-2020 NetFoundry Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <config-utils.h>
+#include <string.h>
+#include <instance.h>
+#include <ziti/ziti_log.h>
+
+// to store the whole tunnel status data
+#define MAX_BUFFER_LEN 1024 * 1024
+#define MIN_BUFFER_LEN 512
+
+static uv_mutex_t mutex;
+
+bool initialize_instance_config() {
+    int mutex_init = uv_mutex_init(&mutex);
+    if (mutex_init != 0) {
+        ZITI_LOG(TRACE, "Could not initialize resources for the config, config file will not be updated");
+    }
+    return mutex_init;
```

Comment:
mutex_init is an int, but this function returns bool. furthermore, uv_mutex_init returns 0 on success, which is `false` when interpreted as a boolean.

this function should either return the uv status code directly (as an int), or explicitly map the code to a bool:
```
    return (mutex_init == 0 ? true : false;
```
also, this function's return value is ignored when it is called from `run_tunnel`

Reply:
fixed the code